compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/SymbolTable.java

```diff
@@ -272,6 +303,7 @@ private void defineOperators() {
         defineBinaryOperator(OperatorKind.EQUAL, intType, intType, booleanType, InstructionCodes.IEQ);
         defineBinaryOperator(OperatorKind.EQUAL, byteType, byteType, booleanType, InstructionCodes.BEQ);
         defineBinaryOperator(OperatorKind.EQUAL, floatType, floatType, booleanType, InstructionCodes.FEQ);
+        defineBinaryOperator(OperatorKind.EQUAL, decimalType, decimalType, booleanType, InstructionCodes.DEQ);
```

Comment:
we don't need this right?

Reply:
We need separate instructions for DEQ and DNE as we need to compare the values of bigdecimal for decimal type. Further, same value can have different precisions (eg: 1.0, 1.00). To handle these cases we need separate instructions as REQ and RNE checks only the reference equality.